testsuite/tests/contrib/lkt_syntax/builtin_methods.lkt

```diff
@@ -0,0 +1,11 @@
+class Name : Node implements TokenNode {
```

Comment:
IIUC, this new file is unused during the execution of the `contrib/lkt_syntax` testcase… what about adding a new test in `contrib/lkt_semantic` instead?

Also, the commit message mentions fixing the `structs/foreign_env_md` test, but this testcase already passes, so it’s not clear what is fixed. Maybe you intended to pass `lkt_semantic_checks=True` to its `build_and_run`?

Reply:
Is it really unused? I'm still discovering how the test suite works but I thought that just adding a new file is enough. @raph-amiard, what do you think about the `lkt_syntax` vs `lkt_semantic` folder for this test.

Sorry, the commit message wasn't clear enough. You're right, this test already pass but it doesn't when you use the new in-development version of the langkit language. This task was given to me as an exercice to make me discover how the new LKT frontend is supposed to work, and to reduce the gap between the current working version of langkit and the next one. In that scope, I had to make this test pass with the future version of LKT.

Hope it helps!

Reply:
> Is it really unused? I'm still discovering how the test suite works but I thought that just adding a new file is enough.

No problem. :-) For most (all?) tests, the entry point is the `test.py` file, so https://github.com/AdaCore/langkit/blob/master/testsuite/tests/contrib/lkt_syntax/test.py in this case. The real work happens in the `build_and_run` function (defined here: https://github.com/AdaCore/langkit/blob/master/testsuite/python_support/utils.py#L184), and we can see there that it just compiles the `expected_concrete_syntax.lkt` source file, which leads me to think that `builtin_methods.lkt` is ignored.

> Sorry, the commit message wasn't clear enough. You're right, this test already pass but it doesn't when you use the new in-development version of the langkit language. This task was given to me as an exercice to make me discover how the new LKT frontend is supposed to work, and to reduce the gap between the current working version of langkit and the next one. In that scope, I had to make this test pass with the future version of LKT.

Yep, we’re on the same page about this. :-) Now, even though the transition to Lkt (as opposed to the Python DSL) is still in the works, we try in the testsuite to excercize everything that is implemented. This is why `build_and_run` has the two `types_from_lkt` and `lkt_semantic_checks` arguments: by default we just make sure that the Lkt unparsing (go check with Raphaël and Romain if you don’t know what it is yet) yields what is expected (in `expected_concrete_syntax.lkt`), but these two arguments make it possible to respectively use definitions in the Lkt source file to generate the analysis library or just run semantic checks on that source.

Here, to check that the new builtin declarations you are adding have the intended effect, I’m suggesting to edit the `structs/foreign_env_md` testcase so that semantic checks are enabled, in the hope that these semantic check work on the properties using `.parent` and lexical envs, and thus work with your changes (i.e. they would fail without). We try to do this for every feature we add to Lkt.

Reply:
Thanks for the explanation, I guess it's all good now?